src/components/search/SearchPanel.tsx

```diff
@@ -0,0 +1,233 @@
+    const targetHostNames = conditions.filter((e) => e.type == "username").map((e) => e.value);
+    const targetPostNames = conditions.filter((e) => e.type == "title").map((e) => e.value);
+    SearchPostByConditions(targetTagIds, targetHostNames, targetPostNames, supabaseClient)
+      .then((matchedPosts) => {
+        props.setPosts(matchedPosts);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  const handleSubmit = (value: string) => {
+    if (value.length == 0) {
+      setErrMsg("Give at least one character.");
+      return;
+    }
+    const prefixedRemoved =
+      searchMode === "username" || searchMode === "tag" ? value.slice(1) : value;
+    if (prefixedRemoved.length == 0) {
+      setErrMsg("Give at least one character.");
+      return;
+    }
+    const selectedInSameType = searchResults
+      .filter((e) => e.type == searchMode)
+      .map((e) => e.value);
+    if (selectedInSameType.includes(prefixedRemoved)) {
+      setErrMsg(`You already insert this ${searchMode == "username" ? "host" : "title"} keyword.`);
+      return;
+    }
+    const conditions = [...searchResults, { type: searchMode, value: prefixedRemoved }];
+    setSearchResults(conditions);
+    setSearchMode("title");
+    setSearchText("");
+    updatePosts(conditions);
+  };
+
+  const handleDeleteTag = (index: number) => {
+    const conditions = searchResults.filter((_, i) => i !== index);
+    setSearchResults(conditions);
+    updatePosts(conditions);
+  };
+
+  const getOptions = () => {
+    if (searchText.length == 0 || searchMode != "tag") return [];
+    const options = tagNames.filter((tag) =>
+      tag.toLowerCase().includes(searchText.slice(1).toLowerCase())
+    );
+    const selectedValues = searchResults
+      .filter((result) => result.type === searchMode)
+      .map((result) => result.value);
+    return options.filter((option) => !selectedValues.includes(option)).map((e) => "#" + e);
+  };
+
+  const handleInputValueChange = (value: string) => {
+    setErrMsg("");
+    setSearchText(value);
+    if (value.startsWith("@")) {
+      setSearchMode("username");
+    } else if (value.startsWith("#")) {
+      setSearchMode("tag");
+    } else {
+      setSearchMode("title");
+    }
+  };
+
+  return (
+    <Stack alignItems="center" direction="column">
+      <Autocomplete
+        sx={{ width: "500px" }}
+        disableClearable
+        freeSolo
+        inputValue={searchText}
+        value={searchText}
+        onInputChange={(_, value) => {
+          handleInputValueChange(value);
+        }}
+        onChange={(_, value) => {
+          if (!value) return;
+          handleSubmit(value);
+        }}
+        options={getOptions()}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            name="search"
+            placeholder="Search: @username, #tag, title"
+            variant="outlined"
+            error={errMsg != ""}
+            InputProps={{
```

Comment:
![image](https://user-images.githubusercontent.com/69895436/228531988-8c627a44-3ee5-4c08-9d34-9c6c11f192a2.png)
value and placeholder in textfield should align at center of the box as design
![image](https://user-images.githubusercontent.com/69895436/228532131-930846ce-f483-40e5-a02c-c9f5c9941ec5.png)

Reply:
Done